### DISCUSSION TITLE
How to skip update for Zoom, if Zoom process is currently running?

### DISCUSSION THREAD

**Discussion by MaximSidorchuk** [Romanitho/Winget-AutoUpdate#510] [General]
Hi guys,

I'm just curious about the best strategy to update Zoom app. 
Initially, I don't want to break user experience for anyone. If users are on the Zoom call, and at that moment WAU is triggering the update. So, the users can be disconnected from the call.

So, is it possible to skip Zoom update if Zoom process is running?
But how to achieve this? I checked Mods, but it does not look like the best option for this particular case.

**Comment by KnifMelti** [comment]
Until the branch `user-approval` is a reality, you could create a `Zoom.Zoom-preinstall.ps1` from the `_AppID-template.ps1`.
In that then wait for process to end:
```
#Beginning of Process Name to Wait for to End - optional wildcard (*) after, without .exe, multiple: "proc1","proc2"
$Wait = @("Zoom")
```
**If the user never ends Zoom (AutoStart) that's a problem...**

There's a detection problem in the upgrade process that makes WAU think Zoom is not installed even after a successful upgrade.
`Confirm-Installation.ps1` creates a `InstalledApps.json` that has the Zoom version set to:
```
				{
					"PackageIdentifier" : "Zoom.Zoom",
					"Version" : "> 5.16.10.26186"
				}
```
Because of the `> ` it fails detection @Romanitho (maybe filter out certain characters?)

**Reply by Romanitho** [reply]
I would say that Winget thinks Zoom is not installed 😅

regarding `>` versions, we could skip them as they can cause false fails.

**Reply by Romanitho** [reply]
Maybe we can think about a process to determine if a `>` is present in the version and how to handle it.

**Comment by AndrewDemski-ad-gmail-com** [comment]
> So, is it possible to skip Zoom update if Zoom process is running?

That could be achieved.. Setup has a built-in functionality which could inform users that update must be performed.

TL;DR: You need [proper policy settings](https://support.zoom.com/hc/en/article?id=zm_kb&sysparm_article=KB0065466).

![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/75534654/07b512b9-f8b0-477f-b2d1-4905547a6751)

**Reply by MaximSidorchuk** [reply]
You are right, if we add 2 registry keys as described [here](https://admx.help/?Category=ZoomMeetings&Policy=ZoomMeetings_HKCU::EnableSilentAutoUpdate_Policy) -  then Zoom app will update silently. User will not be apple to disable auto updates in this case, the option will be gray out for him.

**Comment by MaximSidorchuk** [comment]
Thanks, everyone! 
In general, I solved the issue with your help and mods-preinstall functionality.
I added to Zoom.Zoom-preinstall.ps1 logic to check that Zoom is not connected to any meeting, and if so - proceed with the update. Otherwise - skip the update this time.

`
#Mods Function to run PS script block
function RunModsScript ($modsScriptBlock) {
    $RunModsScriptResult = Invoke-Command -ScriptBlock $modsScriptBlock
    return $RunModsScriptResult
}

#Run PowerShell script block in  Zoom.Zoom-preinstall.ps1
$modsScriptBlock = {
    $checkZoomMeetingStatus = if ($zoomProcess = Get-Process -Name Zoom -EA 0) { (Get-NetUDPEndpoint -OwningProcess $zoomProcess.Id -EA 0 | Measure-Object).Count } else { 0 }
    if ($checkZoomMeetingStatus -gt 0) {
        return 'stopZoomUpdate'
    }
}

$ScriptBlockResult = RunModsScript $modsScriptBlock

if ($ScriptBlockResult -eq 'stopZoomUpdate')
{
    Write-ToLog "Meeting is running. Zoom update is skipped."
    continue
}
`

**Reply by Romanitho** [reply]
We all love mods. Thanks @KnifMelti 🥰

**Reply by Romanitho** [reply]
By the way, you can share your mods here:
https://github.com/Romanitho/Winget-AutoUpdate/discussions/categories/mods
